scripts/lib-franklin.js

```diff
@@ -0,0 +1,753 @@
+            value = imgs.map((img) => img.src);
+          }
+        } else if (col.querySelector('p')) {
+          const ps = [...col.querySelectorAll('p')];
+          if (ps.length === 1) {
+            value = ps[0].textContent;
+          } else {
+            value = ps.map((p) => p.textContent);
+          }
+        } else value = row.children[1].textContent;
+        config[name] = value;
+      }
+    }
+  });
+  return config;
+}
+
+/**
+ * Decorates all sections in a container element.
+ * @param {Element} main The container element
+ */
+export function decorateSections(main) {
+  main.querySelectorAll(':scope > div').forEach((section) => {
+    const wrappers = [];
+    let defaultContent = false;
+    [...section.children].forEach((e) => {
+      if (e.tagName === 'DIV' || !defaultContent) {
+        const wrapper = document.createElement('div');
+        wrappers.push(wrapper);
+        defaultContent = e.tagName !== 'DIV';
+        if (defaultContent) wrapper.classList.add('default-content-wrapper');
+      }
+      wrappers[wrappers.length - 1].append(e);
+    });
+    wrappers.forEach((wrapper) => section.append(wrapper));
+    section.classList.add('section');
+    section.dataset.sectionStatus = 'initialized';
+    section.style.display = 'none';
+
+    /* process section metadata */
+    const sectionMeta = section.querySelector('div.section-metadata');
+    if (sectionMeta) {
+      const meta = readBlockConfig(sectionMeta);
+      Object.keys(meta).forEach((key) => {
+        if (key === 'style') {
+          const styles = meta.style
+            .split(',')
+            .map((style) => toClassName(style.trim()));
+          styles.forEach((style) => section.classList.add(style));
+        } else {
+          section.dataset[toCamelCase(key)] = meta[key];
+        }
+      });
+      sectionMeta.parentNode.remove();
+    }
+  });
+}
+
+/**
+ * Updates all section status in a container element.
+ * @param {Element} main The container element
+ */
+export function updateSectionsStatus(main) {
+  // NOTE: added div.section to pick up sections that are listed deep
+  const sections = [
+    ...main.querySelectorAll(':scope > div.section, div.section'),
+  ];
+  for (let i = 0; i < sections.length; i += 1) {
+    const section = sections[i];
+    const status = section.dataset.sectionStatus;
+    if (status !== 'loaded') {
+      const loadingBlock = section.querySelector(
+        '.block[data-block-status="initialized"], .block[data-block-status="loading"]',
+      );
+      if (loadingBlock) {
+        section.dataset.sectionStatus = 'loading';
+        break;
+      } else {
+        section.dataset.sectionStatus = 'loaded';
+        section.style.display = null;
+      }
+    }
+  }
+}
+
+/**
+ * Decorates all blocks in a container element.
+ * @param {Element} main The container element
+ */
+export function decorateBlocks(main) {
+  // NOTE: added deeplyNestedBlocks for support deeply nested blocks
```

Comment:
@snowiewdev do you have an example where we'd need deeply nested blocks?

Reply:
Hey @trieloff @kptdobe, for `decorateBlocks(main)`, the use case is more specific, in which I can copy the function and change to `customDecorateBlocks` in `scripts.js` without changing the boilerplate. Example attached here:

Sidekick documentation page
Site Link: https://main--helix-website--adobe.hlx.page/docs/sidekick
Google Doc: https://docs.google.com/document/d/1EcS0tuq9_7nmqpDotuRY0IOmO3L8oGcbp2QOpRsNHxs/edit

If removing the `deeplyNestedBlocks` selectors, the gist will not be rendered as the embed link is nested deeper so `main.querySelectorAll(`div.section > div > div`)` inside the section with `Section Metadata content` may not be able to pick that up. 

Let me know if the team prefers 
1) keeping the modification in lib-franklin.js as it is in this PR
2) copy the function to `scripts.js` and renamed to `customDecorateBlocks(main)` for custom edit
3) any other better adjustments suggested

Thank you!

Reply:
Again, you mention influence from `Section Metadata`. Looks like a pattern. We should really isolate and illustrate the difference (with and without `Section Metadata`) so it can be addressed in the boilerplate.